src/ripplegw/interfaces.py

```
@@ -0,0 +1,374 @@
+from .waveforms.IMRPhenomXAS_NRTidalv3 import gen_IMRPhenomXAS_NRTidalv3_hphc
+from .waveforms.SineGaussian import gen_SineGaussian_hphc
+from .waveforms.IMRPhenomXPHM import generate_xphm
+from .conversions import Mc_eta_to_ms
+
+
+class Waveform(ABC):
+    def __init__(self):
+        return NotImplemented
+
+    def __call__(
+        self, axis: Float[Array, " n_freq"], params: dict[str, Float]
```

Ruff (0.15.9)

[error] Syntax error in forward annotation: Expected an expression

(F722)